src/HazelcastClient.ts

```diff
@@ -128,26 +132,98 @@ export class HazelcastClient {
         } else {
             this.config = failoverConfig.clientConfigs[0];
         }
+        this.connectionRegistry = new ConnectionRegistryImpl(this.config.connectionStrategy);
         this.failoverConfig = failoverConfig;
+        this.errorFactory = new ClientErrorFactory();
+        this.serializationService = new SerializationServiceV1(this.config.serialization);
         this.instanceName = this.config.instanceName || 'hz.client_' + this.id;
         this.loggingService = new LoggingService(this.config.customLogger,
             this.config.properties['hazelcast.logging.level'] as string);
         this.loadBalancer = this.initLoadBalancer();
-        this.listenerService = new ListenerService(this);
-        this.serializationService = new SerializationServiceV1(this.config.serialization);
-        this.nearCacheManager = new NearCacheManager(this);
-        this.partitionService = new PartitionServiceImpl(this);
-        this.lifecycleService = new LifecycleServiceImpl(this);
+        this.nearCacheManager = new NearCacheManager(
```

Comment:
I am getting test error on `LostInvalidationTest`. Tried to debug, but cant manage to do it. Need help. I think the failure might be related to changes on NearCacheManager.